### PR TITLE
[autoscaler] Speedup startup!

### PR DESCRIPTION
**Pull request by ijrsvt**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ijrsvt** [comment]
cc @wuisawesome How does this look for a first pass at speeding up setup commands?

File: python/ray/autoscaler/_private/updater.py [hunk 1]
```diff
@@ -103,6 +106,21 @@ def __init__(self,
         self.auth_config = auth_config
         self.is_head_node = is_head_node
         self.docker_config = docker_config
+        self.separate_commands = separate_commands
+
+    def _combine_commands(self, command_list,
+                          command_file_prefix="") -> List[str]:
+        if self.separate_commands:
+            return command_list
+        script = "set -x\n"
```
Comment (wuisawesome) [inline python/ray/autoscaler/_private/updater.py:115]: Consider `-exo pipefail`?

**Comment by wuisawesome** [comment]
First pass look reasonable.

Is there a reason we have to write the commands to a file (vs `echo $COMMANDS | bash`)? I feel like we could save a round trip that way.

File: python/ray/autoscaler/_private/updater.py (hunk 1 above)
Comment (ijrsvt) [reply python/ray/autoscaler/_private/updater.py:115]: I'll try that!

**Comment by ijrsvt** [comment]
> First pass look reasonable.
> 
> Is there a reason we have to write the commands to a file (vs `echo $COMMANDS | bash`)? I feel like we could save a round trip that way.

Sounds good! That would be fine!

**Comment by bveeramani** [comment]
## :bangbang: ACTION REQUIRED :bangbang:

We've switched our code formatter from YAPF to Black (see #21311).

To prevent issues with merging your code, here's what you'll need to do:

1. Install Black
```
pip install -I black==21.12b0
```

2. Format changed files with Black
```
curl -o format-changed.sh https://gist.githubusercontent.com/bveeramani/42ef0e9e387b755a8a735b084af976f2/raw/7631276790765d555c423b8db2b679fd957b984a/format-changed.sh
chmod +x ./format-changed.sh
./format-changed.sh
rm format-changed.sh
```

3. Commit your changes.
```
git add --all
git commit -m "Format Python code with Black"
```

4. Merge master into your branch.
```
git pull upstream master
```

5. Resolve merge conflicts (if necessary).

After running these steps, you'll have the updated `format.sh`.

**Pull request closed without merging**